microproxy/proxy.py

```diff
@@ -86,28 +87,28 @@ def next_layer(self, current_layer, context):
 
 
 class ProxyServer(tcpserver.TCPServer):
-    def __init__(self, config, proxy_server_handler=None):
-        super(ProxyServer, self).__init__()
+    def __init__(self, config, **kwargs):
+        super(ProxyServer, self).__init__(**kwargs)
         self.config = config
-        self.host = config["host"]
-        self.port = config["port"]
         self.interceptor = Interceptor(config=config)
+        self.layer_manager = LayerManager(config)
+        self.event_manager = EventManager(config, self)
 
     def _handle_connection(self, connection, address):
         try:
             stream = MicroProxyIOStream(connection,
                                         io_loop=self.io_loop,
                                         max_buffer_size=self.max_buffer_size,
                                         read_chunk_size=self.read_chunk_size)
-            future = self.handle_stream(stream, address)
+            future = self.handle_stream(stream)
             if future is not None:
                 self.io_loop.add_future(future, lambda f: f.result())
         except Exception as e:
             logger.exception(e)
             raise
 
     @gen.coroutine
-    def handle_stream(self, stream, port):
+    def handle_stream(self, stream):
         try:
             layer_manager = LayerManager(self.config)
```

Comment:
We have created another layer_manager within this function scope.
This is the commit that I change when removing the blinker.